lib/bolt/plugin.rb

```diff
@@ -288,16 +313,55 @@ def resolve_single_reference(reference)
 
       validate_proc.call(reference)
 
-      begin
+      result = begin
         # Evaluate the plugin and then recursively evaluate any plugin returned by it.
         hook.call(reference)
       rescue StandardError => e
         loc = "resolve_reference in #{plugin_name}"
         raise PluginError::ExecutionError.new(e.message, plugin_name, loc)
       end
+
+      if cache?(reference)
+        cache = read_cache
+        cache.merge!({ id => { 'result' => result,
```

Comment:
```suggestion
        cache.merge!({ id => { 'result' => result,
```

This saves the result in plaintext, which will expose any sensitive information returned by the plugin. Is that something we're concerned about?

Reply:
Yes, I have a note in the commit message to encrypt the cache as a next step. We can do that as part of this iteration if it seems unusable without it, but I figured this was a good mvp.

Reply:
Ah, missed that part, my bad. 😅  I think having it as a next step is okay since it's experimental + opt-in.

Reply:
Np np - it definitely *is* unusable for some users without that.